PrestaShop/PrestaShop
# Fix missing AdminSecurity for 8.1.x

| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | 8.1.x
| Description?      | Before publishing 8.1.0 Release Candidate, I fix some missing `@AdminSecurity` annotations to secure Symfony controllers. See below for details
| Type?             | improvement
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | Please see below. Also we need CI ✅ & [auto tests](https://github.com/boherm/ga.tests.ui.pr/actions/runs/5002456044) ✅
| Fixed ticket?     | Fixes #32575 
| Related PRs       | 
| Sponsor company   | 

## Added `AdminSecurity()` annotations

### New API controllers

Controllers
- Configure/AdvancedParameters/AuthorizationServer/ApiAccessController.php
- Configure/AdvancedParameters/AuthorizationServer/ApplicationController.php
were not secured. I fix this. But they are not implemented so it was not a real problem. No need to test, no impact.

### Backoffice translations page

The controller action in charge of rendering the page (other actions are done by ajax endpoints connected to a VueJS application) was not secured. I fix this.

### Backoffice product page v1

Some ProductController actions for the listing were not secured. I fix this.

Multiple ProductImageController.php actions, used to edit & upload images in the backoffice Product Page V1, were not secured. This is because they receive and return JSON, and `AdminSecurity()` does not handle JSON. But leaving these unprotected is worse, so it's better to add the annotation to secure them. Same for VirtualProductController.

### Unfinished Symfony pages

Some Symfony pages not yet completed (= work in progress) were not secured. Annotation was added to them although they have no impact, it's a nice-to-have. Examples: CountryController, CarriersController. No need to test, no impact.

## How to test

Please verify that:

1.  The "translations" backoffice symfony page
- can be accessed by users which are granted correct READ access [through the backoffice Permissions page](https://docs.prestashop-project.org/1.7-documentation/user-guide/configuring-shop/advanced-parameters/team/permissions)
- cannot be accessed by users without these rights
2. The backoffice product page v1
- can be accessed by users which are granted correct READ access [through the backoffice Permissions page](https://docs.prestashop-project.org/1.7-documentation/user-guide/configuring-shop/advanced-parameters/team/permissions)
- can be used (= create, edit) by users which are granted correct WRITE access through the backoffice Permissions page
- cannot be accessed by users without these rights
